File: DP/py/memory.py


memo = {}

def sum_n(n):
 
 if n in memo:
   print("second")
   return memo[n]
 else:
   print("first")
   sum = 0
   for i in range(n+1):
     sum += i
   memo[n] = sum

 return sum


print(sum_n(5))
print(sum_n(5))
   
    

File: DP/js/memory.js
const mem = [];

function sum(n,v) {
  if (mem[n]) {
    console.log({v});
    return mem[n];
  } else {
    console.log({v});
    let sum = 0;

    for (let index = 0; index <= n; index++) {
      sum += index;
    }

    mem[n] = sum;
    return sum;
  }
}

console.log(sum(5,1))
console.log(sum(5,2))
